tests/unit_tests/restruct_poc/system/test_integrate_residual_vector.cpp

```diff
@@ -0,0 +1,254 @@
+#include <Kokkos_Core.hpp>
+#include <gtest/gtest.h>
+
+#include "test_integrate_matrix.hpp"
+
+#include "src/restruct_poc/beams/beams.hpp"
+#include "src/restruct_poc/system/integrate_residual_vector.hpp"
+#include "src/restruct_poc/types.hpp"
+#include "tests/unit_tests/restruct_poc/test_utilities.hpp"
+
+namespace openturbine::restruct_poc::tests {
+
+void TestIntegrateResidualVector_OneElementOneNode_all_zero() {
+    constexpr int num_elements = 1;
+    constexpr int num_nodes = 1;
+    auto node_state_indices = get_node_state_indices<num_elements, num_nodes>();
+
+    auto node_FE = Kokkos::View<double[num_elements * num_nodes][6]>("node_FE");
+    auto node_FI = Kokkos::View<double[num_elements * num_nodes][6]>("node_FI");
+    auto node_FG = Kokkos::View<double[num_elements * num_nodes][6]>("node_FG");
+    auto node_FX = Kokkos::View<double[num_elements * num_nodes][6]>("node_FX");
+
+    auto residual_vector = Kokkos::View<double[num_elements * num_nodes * 6]>("residual_vector");
+
+    Kokkos::parallel_for(
+        num_elements * num_nodes,
+        IntegrateResidualVector{
+            node_state_indices, node_FE, node_FI, node_FG, node_FX, residual_vector}
+    );
+
+    expect_kokkos_view_1D_equal(residual_vector, {0., 0., 0., 0., 0., 0.});
+}
+
+TEST(IntegrateResidualVector, OneElementOneNode_all_zero) {
+    TestIntegrateResidualVector_OneElementOneNode_all_zero();
+}
+
+template <int n_elem, int n_nodes>
+auto get_node_vector(
+    std::string_view name, const std::array<double, n_elem * n_nodes * 6>& vector_data
+) {
+    using VectorView = Kokkos::View<double[n_elem * n_nodes][6]>;
+    using HostVectorView = Kokkos::View<const double[n_elem * n_nodes][6], Kokkos::HostSpace>;
+    auto shape = VectorView(std::string{name});
+    auto host_shape = Kokkos::create_mirror(shape);
+    auto shape_data_view = HostVectorView(vector_data.data());
+    Kokkos::deep_copy(host_shape, shape_data_view);
+    Kokkos::deep_copy(shape, host_shape);
+    return shape;
+}
+
+template <int n_elem, int n_nodes>
+auto get_node_FE(const std::array<double, n_elem * n_nodes * 6>& vector_data) {
+    return get_node_vector<n_elem, n_nodes>("node_FE", vector_data);
+}
+
+template <int n_elem, int n_nodes>
+auto get_node_FI(const std::array<double, n_elem * n_nodes * 6>& vector_data) {
+    return get_node_vector<n_elem, n_nodes>("node_FI", vector_data);
+}
+
+template <int n_elem, int n_nodes>
+auto get_node_FG(const std::array<double, n_elem * n_nodes * 6>& vector_data) {
+    return get_node_vector<n_elem, n_nodes>("node_FG", vector_data);
+}
+
+template <int n_elem, int n_nodes>
+auto get_node_FX(const std::array<double, n_elem * n_nodes * 6>& vector_data) {
+    return get_node_vector<n_elem, n_nodes>("node_FX", vector_data);
+}
```

Comment:
Should these functions live in the hpp file as done for the previous tests?

Reply:
I chose not to keep them in the .hpp file since they are only used locally (not reused by any tests that live in other files).

Reply:
Sounds good, just making sure.